tutorials/diy-module.md

```diff
@@ -0,0 +1,390 @@
+
+The other power connection is ground, which we'll connect to one side of our button. It doesn't matter which side, because a button is just a momentary switch that creates and breaks a connection. You can't hook it up backward. We'll get to connecting the other side of the button in a minute.
+
+### Communication
+As mentioned above, normally your communication protocol is determined by your module. In the case of a button, we use a digital GPIO pin because we want to read the state of the button.
+
+Each port on the Tessel has [several digital I/O pins that can serve this purpose](https://tessel.io/docs/hardwareAPI#pin-mapping), and you are free to pick any one you like because it doesn't matter.
+
+We're going to choose port A's pin 1, which is what we will hook up to the other side of the button. When the button is not pressed, our input pin will read high, or true. When we press it, we are making a connection between our GPIO pin and ground, which will cause a low state to be present on the input pin. This is what the design looks like.
+
+<h1 style="text-align:center;"><img src="https://s3.amazonaws.com/technicalmachine-assets/tutorials/diy-module-guide/switch_schematic.png" /></h1>
+
+<p style="text-align:center"><em>Schematic of button connections</em></p>
+
+<h1 style="text-align:center;"><img src="https://s3.amazonaws.com/technicalmachine-assets/tutorials/diy-module-guide/button_module_angle.jpg" width=300 /></h1>
+
+<p style="text-align:center"><em>Soldered together on a proto-module board.</em></p>
+
+Don't let the soldering part scare you. Soldering components like this onto a proto-module is a little harder than learning to use a hot glue gun, but not a lot harder.
+[This tutorial from Sparkfun](https://learn.sparkfun.com/tutorials/how-to-solder---through-hole-soldering) is a great place to start learning soldering.
+
+Let's go ahead and plug our module into Port A. **Remember to never make connections while your Tessel is powered up. Always unplug it first.**
+
+### Software
+With everything hooked up, it's time to write some Javascript.
+
+Actually, we're going to reuse the code from Kelsey and modify it just a bit. Since she followed the style guidelines and shared her work on NPM, we actually don't have to write the bulk of the code. She's even provided a [Quick Start guide](https://github.com/Frijol/tessel-button#quick-start-i-just-want-a-button) in her documentation, so we'll use that.
+
+  1. Install the tessel-gpio-button package. This will allow us to reuse Kelsey's work.
+
+    ```npm install tessel-gpio-button```
+
+  2. Create a file named **myButton.js** and copy her Quick Start code into it. It should look like this:
+```js
+    // examples/button.js
+    // Count button presses
+
+    var tessel = require('tessel');
+    var buttonLib = require('../');
+    var myButton = buttonLib.use(tessel.port['GPIO'].pin['G3']);
```

Comment:
read carefully - this is the direct copy/paste of my tessel-button repo; he actually calls it out as needing to change in the directions below